### ISSUE TITLE
Python wheel does not get updated on a Databricks cluster

### ISSUE THREAD

**Issue by tanya-borisova** [UCLH-Foundry/FlowEHR#197]
When working in a persistent environment, sometimes building a new Python wheel and updating it in DBFS does not result in the new library getting picked up on Databricks cluster if the cluster is already running.
This seems to be a result of caching that occurs on Databricks cluster.

Known mitigation is to terminate the cluster and restart the data pipeline job.

I think tainting the Databricks cluster terraform resource each time we deploy a pipeline should work as a fix.

**Comment by anastasiakuzn** [comment]
@tanya-borisova to add notes.

**Comment by tanya-borisova** [comment]
I have tried to taint a resource in Terraform so that the cluster gets recreated by Terraform each time, but this didn't work - the old version of the code was still sometimes getting picked up by Terraform. 

I think the only way to properly fix this is to ensure that each new version of the library has a different name. For example, we could implement the versioning proposal (https://github.com/UCLH-Foundry/FlowEHR-Data-Forge/issues/31), or a simpler version of it, when we always add a hash the path of any wheel uploaded to DBFS, and then check that it doesn't exist already before uploading.

**Comment by damoodamoo** [comment]
if the library had a different name, does that get picked up by the running cluster without needing to restart / destroy it?

**Comment by tanya-borisova** [comment]
Yes, it does. Same with setting a different semantic version of the library in toml file, this also works.

**Comment by damoodamoo** [comment]
that sounds good then - definitely want to avoid restarting  / rebuilding the cluster so that's good news. could using the name / toml version to indicate a change of code also mitigate #261 , rather than using the md5 hash?

**Comment by tanya-borisova** [comment]
Terraform uses md5 internally to see whether a file has changed, and this behaviour cannot be modified by the user. 

What specifically in #261 you feel needs to be mitigated?

**Comment by damoodamoo** [comment]
yea this sounds a good idea, and mitigates both issues then, nice!

**Comment by jjgriff93** [comment]
@tanya-borisova did this get fixed? I can't remember

**Comment by tanya-borisova** [comment]
No, this one isn't fixed unfortunately. I wanted to implement the versioning proposal for this, and suffix each new wheel with a version number, but it didn't get implemented.
